### ISSUE TITLE
 FTBS develop branch on Debian/Testing with gcc 7.2.0

### ISSUE THREAD

**Issue by Trophime** [feelpp/feelpp#1060]
Try to rebuild feel from scratch on debian/Testing I run into this error:
```
/home/LNCMI-G/trophime/feelpp/feel/feelalg/staticcondensation.hpp:585:1: error: prototype for ‘void Feel::StaticCondensation<T>::condense(const boost::shared_ptr<Feel::StaticCondensation<T> >&, E&, M_ptrtype&, V_ptrtype&, std::enable_if_t<(decltype (boost::hana::size(e.functionSpace()))::value == 3)>*)’ does not match any in class ‘Feel::StaticCondensation<T>’
 StaticCondensation<T>::condense( boost::shared_ptr<StaticCondensation<T>> const& rhs, E& e, M_ptrtype& S, V_ptrtype& V,
 ^~~~~~~~~~~~~~~~~~~~~
In file included from /home/LNCMI-G/trophime/feelpp/feel/feelalg/staticcondensation.cpp:26:0:
/home/LNCMI-G/trophime/feelpp/feel/feelalg/staticcondensation.hpp:94:10: error: candidates are: template<class T> template<class E, class M_ptrtype, class V_ptrtype> void Feel::StaticCondensation<T>::condense(const boost::shared_ptr<Feel::StaticCondensation<T> >&, E&, M_ptrtype&, V_ptrtype&, std::enable_if_t<(decltype (boost::hana::size(e.functionSpace()))::value == 4)>*)
     void condense( boost::shared_ptr<StaticCondensation<T>> const& rhs, E& e, M_ptrtype& S, V_ptrtype& V,
          ^~~~~~~~
/home/LNCMI-G/trophime/feelpp/feel/feelalg/staticcondensation.hpp:90:10: error:                 template<class T> template<class E, class M_ptrtype, class V_ptrtype> void Feel::StaticCondensation<T>::condense(const boost::shared_ptr<Feel::StaticCondensation<T> >&, E&, M_ptrtype&, V_ptrtype&, std::enable_if_t<(decltype (boost::hana::size(e.functionSpace()))::value == 3)>*)
     void condense( boost::shared_ptr<StaticCondensation<T>> const& rhs, E& e, M_ptrtype& S, V_ptrtype& V,
          ^~~~~~~~
....
```

**Comment by Trophime** [comment]
```boost``` version is 1.62.0

**Comment by vincentchabannes** [comment]
This compilation error can be fix by including the method definition directly at  the declaration. But it's strange, it seems to be a bug of gcc with ```std::enable_if``` when it used with a separated declaration/definition of the method.

**Comment by prudhomm** [comment]
This bug occurs also with gcc 7.3. I am adding a fix for the v0.104.0 release

**Comment by prudhomm** [comment]
This bug breaks debian/ubuntu package builds
